sw/device/mask_rom/boot.md

```diff
@@ -1,71 +1,134 @@
-# Pseudo-code for Secure Boot Process.
+# Pseudo-code for Secure Boot Process
 
-This file should be read in conjunction with secure boot specification. Some
-references to that document are included.
+This file should be read in conjunction with secure boot specification.
+References to that document are included.
 
+1. Power on (entirely in hardware)
 
-1. Power on (entirely in hardware) Open Q: Whether SW has to configure PMP
-  initial region.
+   **Open Q:** Whether SW has to configure PMP initial region.
 
 2. Execution Begins
-  - CRT Startup Code (ASM)
-    - Clean Slate (SRAM, Registers)
-    - Well-defined interrupt/exception handler
-        Open Q: Do we want to enable interrupts in mask rom?
-    - Jump into C code (`boot`):
+   *   CRT Startup Code (Written in Assembly)
+
+       *   Clean Device State Part 1. (Registers, main SRAM)
+
+       *   Well-defined interrupt/exception handler.
+
+           **Open Q:** Do we want to enable interrupts in mask rom?
+
+       *   Setup structures for C execution (`.data` & `.bss` sections, stack).
+
+       *   Jump into C code.
+
+   *   Main Secure Boot Software (Written in C)
+       *   Orchestrated from `boot`:
 
 ```c
 void boot(void) {
-  // - Clean all other hardware state (using DIF Reset functions)
+  // - Clean Device State Part 2
   clean_device_state();
 
+  // - Enable Memory Protection
+  //   - SRAM Scramble
```

Comment:
This potentially has to happen before clearing SRAM, otherwise SRAM with all physical-zeros will appear as all-scrambled. Maybe just put a parenthetical here that it might need to happen earlier.

Reply:
Done